src/Tasks.UnitTests/AssemblyDependency/Miscellaneous.cs

```diff
@@ -3686,6 +3686,37 @@ public void ConflictGeneratesMessageReferencingAssemblyName()
             warningMessage.ShouldContain(ResourceUtilities.FormatResourceStringIgnoreCodeAndKeyword("ResolveAssemblyReference.FourSpaceIndent", ResourceUtilities.FormatResourceStringIgnoreCodeAndKeyword("ResolveAssemblyReference.ReferenceDependsOn", "D, Version=1.0.0.0, CulTUre=neutral, PublicKeyToken=aaaaaaaaaaaaaaaa", Path.Combine(s_myLibraries_V1Path, "D.dll"))));
         }
 
+        [Fact]
+        public void ConflictOutputsExtraInformationOnDemand()
```

Comment:
Should we make this a theory and test what happens when the item has the `.dll` extension? @dsplaisted is there a concrete answer for including `.dll` in the assembly name? Or did your comment, "possibly with .dll possibly not," refer to the fact that `assemblyName.Name` may not have the extension along with it?

Reply:
I understood him to mean that the SDK can handle the output having the .dll extension or not having it—it's the same information, just presented slightly differently. Since I'm creating the item, and I went with no .dll extension, that shouldn't be an issue.

Reply:
I wasn't sure whether the assembly name processed by ResolveAssemblyReference would have the .dll extension on it.  If it consistently does not have the extension, then I would leave it out.  It's also theoretically possible to reference an .exe, I believe.

Reply:
The summary behind `assemblyName.Name`:
> Gets or sets the simple name of the assembly. This is usually, but not necessarily,
the file name of the manifest file of the assembly, minus its extension.

Looks like the SDK will have to account for both scenarios, unless there's a preference on how we handle it here. I'm okay with either scenario.

Reply:
Sounds like MSBuild shouldn't do anything special here then (ie don't add the .dll extension).